tests/unit_tests/test_server_client.py

```diff
@@ -574,13 +582,24 @@ async def test_ainvoke(async_remote_runnable: RemoteRunnable) -> None:
     # Test tracing
     tracer = FakeTracer()
     assert await async_remote_runnable.ainvoke(1, config={"callbacks": [tracer]}) == 2
-    assert len(tracer.runs) == 1
-    # Light test to verify that we're picking up information about the server side
-    # function being invoked via a callback.
-    assert tracer.runs[0].child_runs[0].name == "RunnableLambda"
-    assert (
-        tracer.runs[0].child_runs[0].extra["kwargs"]["name"] == "add_one_or_passthrough"
-    )
+    # Picking up the run from the server side, and client side should also log a run
```

Comment:
This looks like funkiness created by the fact that we're testing the client and the server in the same process AND using contextvars in langchain-core to propagate the config (so looks like a test only artifact)